assembly/runtime.ts

```diff
@@ -677,37 +669,54 @@ export abstract class BaseContext {
   onDone_: (thiz: BaseContext) => bool;
   onDelete_: (thiz: BaseContext) => void;
 
-  constructor() {
+  constructor(context_id_: u32) {
+    this.context_id = context_id_;
```

Comment:
All contexts should have an id. Just enforcing it this here and avoiding code duplication.